### PR TITLE
Lars/nodecode

### PR DESCRIPTION
**Pull request by Lars-Thestorf**

### CODE REVIEW

File: doorbell.ino [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+      Serial.println(" Seconds");
+      digitalWrite(mutePin,HIGH);
+      timerAutoOpen.once(timer, endAutoOpen);
+    }
+    if (request.indexOf("/reset") != -1)  {
+      endAutoOpen();
+      timerAutoOpen.detach();
+    }
+   
+    // Return the response
```
Comment (blixtra) [inline doorbell.ino]: Please construct this first then do one Print call

File: doorbell.ino [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+  connectWiFi();
+ 
+  // Print the IP address
+  Serial.print("Use this URL to connect: ");
+  Serial.print("http://");
+  Serial.print(WiFi.localIP());
+  Serial.println("/");
+
+  WiFiCheckTimer.attach(10,checkWiFi);
+  attachInterrupt(digitalPinToInterrupt(ringPin),ringInterupt,RISING);
```
Comment (blixtra) [inline doorbell.ino]: Space between args

Comment (blixtra) [inline doorbell.ino]: Please do that everywhere.

File: doorbell.ino [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+               "Connection: close\r\n\r\n");
+      delay(10);
+ 
+      // Read all the lines of the reply from server and print them to Serial
+      while(client.available()){
+        String line = client.readStringUntil('\r');
+        Serial.print(line);
+      }
+ 
+      Serial.println();
```
Comment (blixtra) [inline doorbell.ino]: But a new line on the next line instead of having to Prints

File: doorbell.ino [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+   
+    // Match the request
+    if (request.indexOf("/door") != -1)  {
+      digitalWrite(openPin, HIGH);
+      timerDoor.once_ms(opentime, releaseDoor);
+    }
+    if (request.indexOf("/autoopen=") != -1)  {
+      int sub1 = request.indexOf("/autoopen=")+10;
+      int sub2 = request.indexOf(" ",sub1+1);
+      String sub = request.substring(sub1,sub2);
```
Comment (blixtra) [inline doorbell.ino]: What happens if you have something other than an int?

File: doorbell.ino [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+      int timer = sub.toInt()*60;
+      autoOpen=true;
+      Serial.print("In Auto-Open Mode for ");
+      Serial.print(timer);
+      Serial.println(" Seconds");
+      digitalWrite(mutePin,HIGH);
```
Comment (blixtra) [inline doorbell.ino]: Setting state and flipping pins should be done in its own function(s). Please break those out.

File: doorbell.ino (hunk 4 above)
Comment (Lars-Thestorf) [inline doorbell.ino]: Arduino handles everything that is not a string as a 0

File: doorbell.ino [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+  Serial.println(ssid);
+ 
+  WiFi.begin(ssid, password);
+  WiFi.config(staticIP, gateway, subnet);
+ 
+  while (WiFi.status() != WL_CONNECTED) {
+    delay(500);
+    Serial.print(".");
+  }
+  Serial.println("");
```
Comment (blixtra) [inline doorbell.ino]: Same issue here

File: doorbell.ino [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+  }
+  if (trys<10){
+    // Read the first line of the request
+    String request = client.readStringUntil('\r');
+    Serial.println(request);
+    client.flush();
+   
+    // Match the request
+    if (request.indexOf("/door") != -1)  {
+      digitalWrite(openPin, HIGH);
```
Comment (blixtra) [inline doorbell.ino]: Move this logic into a function please.

File: doorbell.ino [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+#include <Ticker.h>
+ 
+const char* ssid = "SSID";
+const char* password = "PASS";
+IPAddress staticIP(192,168,35,22);
+IPAddress subnet(255,255,255,0);
+IPAddress gateway(192,168,35,107);
+IPAddress host(192,168,35,112);
+
+const int opentime = 2000;
```
Comment (blixtra) [inline doorbell.ino]: Is this how long the buzzer will sound? If so, lease make it at least 4 sec.

Comment (Lars-Thestorf) [inline doorbell.ino]: @blixtra This is how long the button will act as pressed
the buzzer will always sound 6 seconds doesn't matter how long you press it

Comment (blixtra) [inline doorbell.ino]: :+1:

File: doorbell.ino [hunk 9]
```diff
@@ -0,0 +1,182 @@
...
+ 
+  // Wait until the client sends some data
+  Serial.println("new client");
+  int trys=0;
+  while(!client.available()&&trys<10){
+    delay(1);
+    Serial.println(".");
+    trys++;
+  }
+  if (trys<10){
```
Comment (blixtra) [inline doorbell.ino]: Spaces should separate operators from their neighbors.

File: doorbell.ino [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+    Serial.println(request);
+    client.flush();
+   
+    // Match the request
+    if (request.indexOf("/door") != -1)  {
+      openDoor();
+    }
+    if (request.indexOf("/autoopen=") != -1)  {
+      int sub1 = request.indexOf("/autoopen=")+10;
+      int sub2 = request.indexOf(" ", sub1+1);
```
Comment (blixtra) [inline doorbell.ino]: Like here. Should be `sub1 + 1`

File: doorbell.ino [hunk 11]
```diff
@@ -0,0 +1,182 @@
...
+    delay(500);
+    Serial.print(".");
+  }
+  Serial.println("\r\nWiFi connected");
+  server.begin();
+  Serial.println("Server started");
+}
+
+void releaseRing(){
+  ringval=false;
```
Comment (blixtra) [inline doorbell.ino]: Also here. Let's use spaces on either side of the `=`.

**Pull request merged**